agg-ast/schema_derivation/src/schema_derivation_tests/stage.rs

```diff
@@ -1307,6 +1307,44 @@ mod union {
             ..Default::default()
         })
     );
+    test_derive_stage_schema!(
+        union_pipeline_no_collection,
+        expected = Ok(Schema::Document(Document {
+            keys: map! {
+                "_id".to_string() => Schema::Atomic(Atomic::Integer),
+                "a".to_string() => Schema::Atomic(Atomic::String),
+                "food".to_string() => Schema::AnyOf(set!(
+                    Schema::Atomic(Atomic::String),
+                    Schema::Atomic(Atomic::Integer)
+                )),
+                "out".to_string() => Schema::Atomic(Atomic::Decimal),
+            },
+            required: set!("food".to_string()),
+            ..Default::default()
+        })),
+        input = r#"{"$unionWith": {"pipeline": [{"$documents": [{"_id": 1, "a": "hello world", "food": 1}]}]}}"#,
+        starting_schema = Schema::Document(Document {
+            keys: map! {
+                "food".to_string() => Schema::Atomic(Atomic::String),
+                "out".to_string() => Schema::Atomic(Atomic::Decimal),
+            },
+            required: set!("food".to_string(), "out".to_string()),
+            ..Default::default()
+        })
+    );
+    test_derive_stage_schema!(
+        union_not_enough_args,
+        expected = Err(crate::Error::NotEnoughArguments("$unionWith".to_string())),
+        input = r#"{"$unionWith": {}}"#,
```

Comment:
So did you run into this as an actual pipeline or is this just preventative in case we encounter this in the future? Does mongodb not reject views that contain this type of stage?

Reply:
you are correct that we should never run into this, and pipeline correctness is out of scope for this project. this won't actually run on server! We should never hit this error. I keep waffling on this case by case on if we should simply panic, but given we have an existing descriptive error for this case, it seemed free to me to point out the user needs to add args